legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/sybaseIQExtension.pure

```diff
@@ -237,6 +238,25 @@ function <<access.private>> meta::relational::functions::sqlQueryToString::sybas
     '%s limit %s,%s'->format([$format.separator, $s.fromRow->toOne()->getValueForTake(), $size]);
 }
 
+function <<access.private>> meta::relational::functions::sqlQueryToString::sybaseIQ::getTransformForSplitPartSybaseIQ():LambdaFunction<{String[3]->String[1]}>[1]
+{
+    {p:String[3]
+                | let string = $p->at(0);
+                  let separator = $p->at(1);
+                  let position = $p ->at(2) -> meta::pure::functions::string::parseInteger();
+                  assert($position>0, 'SplitPart supports only index values greater than or equal to 1');
+                  assert($separator->length() > 0,'SplitPart does not support empty separators'); '';
+                  if($position == 1,| getBeforeAndAfterIndexIQ($string, $separator, $position, 'locate('+ $string+', ' + $separator + ', 0)-1', 'LENGTH('+ $string +')') ,| getBeforeAndAfterIndexIQ($string, $separator, $position, 'locate('+ $string+', ' + $separator + ', 0)', '0'));
```

Comment:
logic need refactoring, so that it is easier to understand. Start by creating 2 fns, one for $position == 1 case, and one for other. Then further see, how the sql string construction can be broken down into smaller parts to increase readability.